Lib/test/test_list.py

```diff
@@ -332,5 +332,25 @@ def test_no_memory(self):
         else:
             self.assertNotEqual(rc, -int(signal.SIGSEGV))
 
+    def test_deopt_from_append_list(self):
+        # gh-132011: it used to crash, because
+        # of `CALL_LIST_APPEND` specializetion failure.
+        code = textwrap.dedent("""
+        l = []
+        def lappend(l, x, y):
+            l.append((x, y))
+        for x in range(3):
+            lappend(l, None, None)
+        try:
+            lappend(list, None, None)
+        except TypeError:
+            pass
+        else:
+            assert False
```

Comment:
I would prefer an explicit:

```suggestion
            raise AssertionError
```